sdk/index.ts

```diff
@@ -0,0 +1,273 @@
+            environment, e.g. email address, username, uuid
+     * @param  {{[key:string]:any}} traits? a dictionary of traits to add / update on the identity in
+            Flagsmith, e.g. {"num_orders": 10}
+     * @returns Flags object holding all the flags for the given identity.
+     */
+    getIdentityFlags(identifier: string, traits?: { [key: string]: any }): Promise<Flags> {
+        traits = traits || {};
+        if (this.environment) {
+            return new Promise(resolve =>
+                resolve(this.getIdentityFlagsFromDocument(identifier, traits || {}))
+            );
+        }
+        return this.getIdentityFlagsFromApi(identifier, traits);
+    }
+
+    async update_environment() {
+        this.environment = await this.getEnvironmentFromApi();
+    }
+
+    private async getJSONResponse(
+        url: string,
+        method: string,
+        body?: { [key: string]: any }
+    ): Promise<any> {
+        const headers: { [key: string]: any } = {};
+        if (this.environmentKey) {
+            headers['X-Environment-Key'] = this.environmentKey as string;
+        }
+
+        if (this.customHeaders) {
+            for (const [k, v] of Object.entries(this.customHeaders)) {
+                headers[k] = v;
+            }
+        }
+
+        const data = await retryFetch(
+            url,
+            {
+                method: method,
+                timeout: this.requestTimeoutSeconds || undefined,
+                body: JSON.stringify(body),
+                headers: headers
+            },
+            this.retries,
+            1000,
+            (this.requestTimeoutSeconds || 10) * 1000
+        );
+
+        if (data.status !== 200) {
+            throw new FlagsmithAPIError(
+                `Invalid request made to Flagsmith API. Response status code: ${data.status}`
+            );
+        }
+
+        return data.json();
+    }
+
+    private async getEnvironmentFromApi() {
+        const environment_data = await this.getJSONResponse(this.environmentUrl, 'GET');
+        return buildEnvironmentModel(environment_data);
+    }
+
+    private getEnvironmentFlagsFromDocument() {
+        if (!this.environment) {
+            throw new FlagsmithClientError(
+                'Unable to build identity model when no local environment present.'
+            );
+        }
+
+        return Flags.fromFeatureStateModels({
+            featureStates: getEnvironmentFeatureStates(this.environment),
+            analyticsProcessor: this.analyticsProcessor,
+            defaultFlagHandler: this.defaultFlagHandler
+        });
+    }
+
+    private getIdentityFlagsFromDocument(identifier: string, traits: { [key: string]: any }) {
+        if (!this.environment) {
+            throw new FlagsmithClientError(
+                'Unable to build identity model when no local environment present.'
+            );
+        }
+
+        const identityModel = this.buildIdentityModel(identifier, Object.values(traits));
```

Comment:
I'm not sure how this would work? This would convert {trait_1:"value"} to ["value"]